catalog/transformer/catalog-transformer-pptx/src/main/java/ddf/catalog/transformer/input/pptx/PptxInputTransformer.java

```diff
@@ -0,0 +1,243 @@
+        }
+
+        return transformLogic(input, id);
+    }
+
+    /**
+     * This is a three step process. First, create a FileBackedOutputStream because we need to
+     * consume the stream twice. Once for the injected inputTransformer and once for Apache POI.
+     * Next, extract the metadata with the injected input transformer. And last, use Apache POI
+     * to create the thumbnail.
+     *
+     * @param input
+     * @param id
+     * @return
+     * @throws IOException
+     * @throws CatalogTransformerException
+     */
+    private Metacard transformLogic(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(
+                FILE_BACKED_THRESHOLD)) {
+            try {
+                int c = IOUtils.copy(input, fileBackedOutputStream);
+                LOGGER.debug("copied {} bytes from input stream to file backed output stream", c);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            Metacard metacard = extractInitialMetadata(fileBackedOutputStream.asByteSource()
+                    .openStream());
+
+            extractThumbnail(metacard,
+                    fileBackedOutputStream.asByteSource()
+                            .openStream());
+
+            return metacard;
+        }
+
+    }
+
+    /**
+     * Extract the initial metadata using the injected input transformer.
+     *
+     * @param input
+     * @return
+     * @throws IOException
+     * @throws CatalogTransformerException
+     */
+    private Metacard extractInitialMetadata(InputStream input)
+            throws IOException, CatalogTransformerException {
+        return inputTransformer.transform(input);
+    }
+
+    /**
+     * SlideShowFactory.create() handles the old-style (OLE2) PPT file and the newer OOXML PPTX files. But
+     * the old-style is not well supported (it's not even a part of the normal jars). The only reason I'm
+     * using SlideShowFactotry instead of just creating an XMLSlideShow directly is that SlideShowFactory
+     * performs the tests for password protected files. Also, from the POI docs, there isn't even any
+     * metadata available for OLE2 nor is there a way to generate a thumbnail.
+     * <p/>
+     * Because Apache POI dynamically loads the classes needed to handle a PPTX file, the default
+     * class loader is unable to find the dependencies during runtime. Therefore, the original class
+     * loader is saved, then current class loader is set to this class's class loader, and finally
+     * the original class loader is restored.
+     *
+     * @param metacard
+     * @param input
+     * @throws IOException
+     */
+    private void extractThumbnail(Metacard metacard, InputStream input) throws IOException {
```

Comment:
I'd like to suggest an alternative before calling into POI. There are relatively cheap checks you can do to see if the .pptx has a embedded thumbnail (conceptually open the zip, check for docProps/thumbnail.jpeg). If its not present, then do the rendering.